tirocks/src/option/read.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    fmt::{self, Debug, Formatter},
+    mem::MaybeUninit,
+};
+
+use tirocks_sys::rocksdb_titandb_TitanReadOptions;
+
+use super::OwnedSlice;
+
+pub type ReadTier = tirocks_sys::rocksdb_ReadTier;
+pub type SequenceNumber = tirocks_sys::rocksdb_SequenceNumber;
+
+/// Options that control read operations
+pub struct ReadOptions {
+    raw: rocksdb_titandb_TitanReadOptions,
+    // Storage for iterate_lower_bound, iterate_upper_bound and timestamp.
+    slice_store: Option<Box<[OwnedSlice; 3]>>,
```

Comment:
Why not split it into three fields and give a meaningful name. One field is set, the memory of three fields are allocated.

Reply:
Then it requires three allocations instead of one.

Reply:
I see what you mean and updated.